File: Data Structures & Algorithms/three-integer-sum/submission-2.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<int> sorted = nums;
        sort(sorted.begin(), sorted.end());
        vector<vector<int>> result;
        for (int i = 0; i < sorted.size() - 1; i++) {
            if (i > 0 && sorted[i] == sorted[i - 1]) {
                continue;
            }
            int start = i + 1;
            int end = sorted.size() - 1;
            while (start < end) {
                if (sorted[start] + sorted[end] + sorted[i] == 0) {
                    result.push_back({sorted[start], sorted[end], sorted[i]});
                    while (start < end && sorted[start] == sorted[start+1]) {
                        start++;
                    } 
                    while (start < end && sorted[end] == sorted[end-1]) {
                        end--;
                    } 
                    start++;
                    end--;
                } else if (sorted[start] + sorted[end] + sorted[i] < 0) {
                    start++;
                } else {
                    end--;
                }
            }
        }
        return result;
    } 
};

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashSet = {}
        for letter in s:
            if letter in hashSet:
                hashSet[letter] += 1
            else:
                hashSet[letter] = 1
        for letter in t:
            if letter in hashSet:
                if hashSet[letter] > 0:
                    hashSet[letter] -= 1
                else:
                    return False
            else:
                return False
        result = 0
        for letter in hashSet:
            result += hashSet[letter]
        if result == 0:
            return True
        else:
            return False